lib/compilers/rust.ts

```diff
@@ -122,11 +122,15 @@ export class RustCompiler extends BaseCompiler {
         if (possibleEditions.length > 0) {
             let defaultEdition: undefined | string;
             if (!this.compiler.semver || this.isNightly()) {
-                defaultEdition = '2021';
+                defaultEdition = '2024';
             } else {
                 const compilerVersion = new SemVer(this.compiler.semver);
-                if (compilerVersion.compare('1.56.0') >= 0) {
+                if (compilerVersion.compare('1.85.0') >= 0) {
+                    defaultEdition = '2024';
+                } else if (compilerVersion.compare('1.56.0') >= 0) {
                     defaultEdition = '2021';
+                } else if (compilerVersion.compare('1.31.0') >= 0) {
+                    defaultEdition = '2018';
```

Comment:
This would change old links created for these older versions to now use 2018, where previously nothing was passed and therfore 2015 was previously used (afaik? 1.86 seems to imply that 2015 is still the default unless otherwise stated).


Would this create issues with such old links that would have not defined 2015 explicitly and now would compile with 2018?

Reply:
> This would change old links created for these older versions to now use 2018,

Would it? As far as I understand the code, this only affects which edition will be selected by default for the edition override setting. E. g. for this link https://godbolt.org/z/s3d5qqo3E there is no edition override selected, so even though `beta`’s `defaultEdition` is 2024, this has no effect on this link.

Reply:
Overrides are set to the defaults when they are missing from the State of the link. https://github.com/compiler-explorer/compiler-explorer/blob/main/static/compiler-shared.ts#L57
In old links this information was not stored in the state and as such the default will apply.

However, this has also happened in the past when this feature was first introduced, and so harm was done either way.

So I think this is acceptable.

Reply:
I see. It’s probably better to remove this again then, given that I think it’s pretty unlikely that a lot of people (or even a few people) will use Rust <= 1.55 for new stuff.

Reply:
I removed this in https://github.com/compiler-explorer/compiler-explorer/pull/7649/commits/e186970a92c823d67ce60be662658e16b9cacd2b.